source/fundamentals/auth.txt

```diff
@@ -0,0 +1,384 @@
+
+The mechanisms that you can use with the latest version of MongoDB Community
+Edition are as follows:
+
+* :ref:`Default <default-auth-mechanism>`
+* :ref:`SCRAM-SHA-256 <scram-sha-256-auth-mechanism>`
+* :ref:`SCRAM-SHA-1 <scram-sha-1-auth-mechanism>`
+* :ref:`MONGODB-CR <mongodb-cr-auth-mechanism>`
+* :ref:`MONGODB-AWS <mongodb-aws-auth-mechanism>`
+* :ref:`X.509 <x509-auth-mechanism>`
+
+If you need to authentication using ``Kerberos`` or ``LDAP`` see the
+:docs:`Enterprise Authentication Mechanisms guide <fundamentals/enterprise-auth>`.
+For more information on establishing a connection to your MongoDB cluster,
+read our :doc:`Connection Guide </fundamentals/connection>`.
+
+Specify an Authentication Mechanism
+-----------------------------------
+
+You can specify your authentication mechanism and credentials when connecting
+to MongoDB using either of the following:
+
+- a connection string
+- a ``MongoCredential`` factory method
+
+A **connection string** or **connection uri** provides instructions on how
+to connect and authenticate to your MongoDB cluster. To specify your
+authentication mechanism using a connection string, include your settings in
+your connection string and pass it to the ``MongoClients.create()`` method
+to instantiate your ``MongoClient``. For descriptions of how to use a
+connection string to specify your authentication method, navigate to the
+relevant mechanism section and select the :guilabel:`Connection String` tab.
+
+The ``MongoCredential`` class contains static factory methods that allow you
+to specify your authentication mechanism and credentials. You can use this
+helper class to keep your code organized. If you choose to use the
+``MongoCredential`` helper class, you need to use the
+``MongoClientSettings.Builder`` class to configure your connection settings
+when constructing your ``MongoClient``. For descriptions of how to use the
+``MongoCredential`` class to specify your authentication method, navigate
+to the relevant mechanism section and select the :guilabel:`MongoCredential`
+tab.
+
+For more information on the aforementioned classes and methods, refer to the
+following API documentation:
+
+- :java-sync-api:`MongoClients.create() </com/mongodb/client/MongoClients.html#create(com.mongodb.ConnectionString)>`
+- :java-sync-api:`MongoClient </com/mongodb/client/MongoClient.html>`
+- :java-sync-api:`MongoClientSettings.Builder </com/mongodb/MongoClientSettings.Builder.html>`
+- :java-sync-api:`MongoCredential </com/mongodb/MongoCredential.html>`
+
+Mechanisms
+----------
+
+.. _default-auth-mechanism:
+
+Default
+~~~~~~~
+
+The default authentication mechanism is a fallback setting that instructs
+the driver to negotiate the first authentication mechanism supported by the
+server in the following order of preference:
+
+#. ``SCRAM-SHA-256``
+#. ``SCRAM-SHA-1``
+#. ``MONGODB-CR``
+
+When you use the default authentication mechanism, the driver first attempts
```

Comment:
I don't think this is accurate.  Where did you get this information from?

Reply:
Ah, what's the behavior then? I was working from the spec: https://github.com/mongodb/specifications/blob/master/source/auth/auth.rst#defaults

Reply:
Right.  It's hard to understand, and what happens depends on the server version that the driver is connected to.

It would be accurate to say just that the driver will default to a mechanism depending on the server version it's connected to.  If you want to go into more detail, you can say that it will default to MONOGDB-CR for 3.6 and below, and for 4.0 and up it will default based on what mechanisms the server advertises support for.

The only retry of auth has to do with the speculative authentication support added in 4.4, but I don't think it's worth mentioning that, as it's just an optimization to try and save one round trip to the server by combining the isMaster from the handshake with the saslStart for the auth conversation.

Reply:
I agree that it's easer to understand by mentioning only that it defaults to the mechanisms the server supports, will update.